### DISCUSSION TITLE
UI Feedback

### DISCUSSION THREAD

**Discussion by jamesditrapani** [peering-manager/peering-manager#397] [Ideas]
As part of the v2.0 milestone for Peering Manager we're currently looking to obtain some feedback around the UI and how it can better serve Network Operators. In the current state a lot of the functionality and information is buried below a few clicks that could potentially be brought to the foreground.

In regards to feedback, we'd like to hear any and all ideas from both operators actively using the tool along with those that have used it in a demo capacity. Specifically:

- Is something missing?
- Does something not need to be there?
- Should a particular item be moved/displayed in a different sub-section?
- etc.

**Comment by traderose** [comment]
Hi @jamesditrapani 

We use PM in prodution to serve and manage peering of 20 IXPs that we are member of with 600+ bgp sessions.

Quick look at UI, we think it would be usefull to have the following:

1. It would be nice to have the option to support multiple cofiguration templates per router.
2. poll bgp sessions status to be available under groups, router direct peering sessions and router IX peering sessions.
3. The option to have a page for all direct_bgp sessions
3. bring scheduled tasks to UI rather than crontab and with more flexiblity. Example, if you want to push config to specific list of routers.
4. add support for prefix generation per AS-SET/AS-MACRO 
5. To add communities option under peering session and also to increase community value CHAR field (value | character varying(50) ) capacity to accept more cummuntities, for example, in juniper you can add multiple members under the same community name. One use case for this, if you want to "don't announce routes" to a list of AS numbers over route server, you tag your prefixes will all releveant communities. 
6. To add some custom fields under IX, groups and IX/direct peering session for flexible use of bgp attributes such as AS-PAPTH, LOCAL_PREF, PREFIX_LENGTH, etc with a reference page similar to bgp communties. And also those fileds can be useful in case someone needs to generate communtiy value per his own custom codes(i.e contient, country, city and route type, etc) especially with large scale networks where peering admins deals with hunderds or thousands of communtites).  
7. To have a page under email tab to track the sent emails.

We have more notes but those are the most important ones.

**Reply by jamesditrapani** [reply]
Thank you! Feel free to throw any of your other ideas/points in the discussion, everything counts!

**Reply by mngan** [reply]
> It would be nice to have the option to support multiple cofiguration templates per router.

Just curious how you intend to use multiple configurations templates per router? Is this like a template for direction sessions and a template for IX sessions?

**Reply by traderose** [reply]
Hi @mngan 

Thats good question. Basically I am looking for as much flexibility which will help for scaling. There few cases that I am thinking of:

1. In large scale networks, the template file can easily grow quickly as you move, so having multiple templates that can be merged on config check and deployment will help to breakdown the code. And what you suggested is a good example. 

2. routers with different roles can share some common config while they need different config requirements per role. 

3. right now configuration template is stored in database. But it  would be helpful if this can be extended to support jinja2 "block extends"  just like in Django which I think will give more flexibility especially when you have static contents. 

Thank you
